bftengine/src/bcstatetransfer/BCStateTran.cpp

```diff
@@ -1731,31 +1743,29 @@ bool BCStateTran::onMessage(const CheckpointSummaryMsg *m, uint32_t msgLen, uint
   // if we have too many messages from the same replica
   if (numOfMsgsFromSender >= (psd_->getMaxNumOfStoredCheckpoints() + 1)) {
     LOG_WARN(logger_, "Too many messages from replica: " << KVLOG(replicaId, numOfMsgsFromSender));
-    return true;
+    return;
   }
 
-  auto p = summariesCerts.find(m->checkpointNum);
-  CheckpointSummaryMsgCert *cert = nullptr;
-
-  if (p == summariesCerts.end()) {
-    cert = new CheckpointSummaryMsgCert(
-        replicaForStateTransfer_, replicas_.size(), config_.fVal, config_.fVal + 1, config_.myReplicaId);
-    summariesCerts[m->checkpointNum] = cert;
-  } else {
-    cert = p->second;
+  if (summariesCerts.find(m->checkpointNum) == summariesCerts.end()) {
+    summariesCerts.emplace(
```

Comment:
Did you forget, or didn't want ot do it as part of your task?
summariesCerts should be re-written (we can discuss this, and add a future task for this):
`map<uint64_t, std::unique_ptr<CheckpointSummaryMsgCert, someDeleter>> summariesCerts;`
So here we would like to create a shared_ptr to make sure there are no leaks.

Reply:
summariesCerts does hold a unique_ptr. Anyway, added a deleter.